app/controllers/kaui/admin_tenants_controller.rb

```diff
@@ -310,9 +310,9 @@ def upload_plugin_config
     plugin_properties = params[:plugin_properties]
     plugin_type = params[:plugin_type]
 
-    plugin_config = Kaui::AdminTenant.format_plugin_config(plugin_name, plugin_type, plugin_properties)
```

Comment:
If it's not too hard, could you write more tests for this as well (since the naming of plugin key vs name is quite confusing, it's not always obvious which is which where)? Again, I think simple unit tests are fine (no need for functional or integration tests).

Reply:
Circling back on this after playing with it locally, we definitively need more tests around this as I don't believe the logic is accurate yet.

If I enter `pay`, `paypal` is suggested (expected, as it matches the `pluginKey` from `plugins_directory.yml`). However, the request `POST http://127.0.0.1:8080/1.0/kb/tenants/uploadPluginConfig/killbill-paypal` is sent to Kill Bill, which is wrong: it should be `http://127.0.0.1:8080/1.0/kb/tenants/uploadPluginConfig/killbill-paypal-express` (see https://github.com/killbill/killbill-paypal-express-plugin#configuration).

There is a similar issue with FirstData: `firstdata_e4` is suggested but `POST http://127.0.0.1:8080/1.0/kb/tenants/uploadPluginConfig/killbill-firstdata_e4` is sent instead of `http://127.0.0.1:8080/1.0/kb/tenants/uploadPluginConfig/killbill-firstdata-e4` (https://github.com/killbill/killbill-firstdata-e4-plugin#configuration).

It seems to work for Braintree (`braintree` -> `braintree_blue` -> `killbill-braintree_blue`) and Adyen (`adyen` -> `adyen` -> `killbill-adyen`) though.

I think all of the entries which have a non-standard naming scheme (i.e. if they have the `artifact_id` key specified in https://github.com/killbill/killbill-cloud/blob/master/kpm/lib/kpm/plugins_directory.yml) need some form of special handling unfortunately.

I just saw your suggestion in the PR description to update the key name in `plugins_directory.yml`, but this has potentially lots of ramifications at this point (code but also docs), so I'd like to avoid it. If there is no easy programmatic way to handle these few snowflakes, just hardcode a special mapping list and create a GitHub issue so we can track it.